core/marc.py

```diff
@@ -710,14 +708,15 @@ def records(
                         work, annotator, force_refresh, self.integration
                     )
                     if record:
-                        this_batch.write(record.as_marc())
-                this_batch_size += pagination.this_page_size
-                if this_batch_size >= upload_batch_size:
+                        record_bytes = record.as_marc()
+                        this_batch_bytes += len(record_bytes)
+                        this_batch.write(record_bytes)
+                if this_batch_bytes >= self.MINIMUM_UPLOAD_BATCH_SIZE_BYTES:
```

Comment:
Minor: It might clearer to just directly use the length of the `BytesIO` object rather then track it separately in `this_batch_bytes`. Something like:

```python
if this_batch.getbuffer().nbytes >= self.MINIMUM_UPLOAD_BATCH_SIZE_BYTES:
```